### PR TITLE
Openexr 2.4.0

### PR DESCRIPTION
**Pull request by madebr**
Specify library name and version:  **openexr/2.4.0**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

openexr 2.3.0 has different cmake configuration variables, so I needed to create a new 2.4.0 subdirectory.

This should close #204 , as that one stalled and has merge conflicts

### CODE REVIEW

File: recipes/openexr/2.4.0/conandata.yml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+sources:
+  "2.4.0":
+    url: "https://github.com/openexr/openexr/archive/v2.4.0.tar.gz"
+    sha256: "4904c5ea7914a58f60a5e2fbc397be67e7a25c380d7d07c1c31a3eefff1c92f1"
+patches:
+  "2.4.0": []
```
Comment (uilianries) [inline recipes/openexr/2.4.0/conandata.yml]: nice, we should adopt it by default

File: recipes/openexr/2.4.0/conandata.yml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+sources:
+  "2.4.0":
+    url: "https://github.com/openexr/openexr/archive/v2.4.0.tar.gz"
+    sha256: "4904c5ea7914a58f60a5e2fbc397be67e7a25c380d7d07c1c31a3eefff1c92f1"
+patches:
```
Comment (uilianries) [inline recipes/openexr/2.4.0/conandata.yml]:
```suggestion
```

as this PR is to a very specific version, including the folder, and you don't need a patch, so you can remove it.

File: recipes/openexr/2.4.0/conandata.yml (hunk 1 above)
Comment (madebr) [reply recipes/openexr/2.4.0/conandata.yml]: Actually this was a leftover after splitting 2.3.0 and 2.4.0 :smiley:

File: recipes/openexr/2.4.0/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+                  "Magic for use in computer imaging applications."
+    topics = ("conan", "openexr", "hdr", "image", "picture")
+    license = "BSD-3-Clause"
+    homepage = "https://github.com/openexr/openexr"
+    url = "https://github.com/conan-io/conan-center-index"
+    settings = "os", "compiler", "build_type", "arch"
+    options = {"shared": [True, False], "fPIC": [True, False]}
+    default_options = {"shared": False, "fPIC": True}
+    generators = "cmake", "cmake_find_package"
+    exports_sources = ["CMakeLists.txt", "patches/*.patch"]
```
Comment (uilianries) [inline recipes/openexr/2.4.0/conanfile.py]:
```suggestion
    exports_sources = ["CMakeLists.txt"]
```

File: recipes/openexr/2.4.0/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+        if self.settings.os == "Windows":
+            self.options.remove("fPIC")
+
+    def requirements(self):
+        self.requires("zlib/1.2.11")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename("openexr-{}".format(self.version), self._source_subfolder)
+        for p in self.conan_data["patches"][self.version]:
```
Comment (madebr) [inline recipes/openexr/2.4.0/conanfile.py]: Move to `self._patch_files`

**Comment by madebr** [comment]
@uilianries Removed

**Comment by uilianries** [comment]
@madebr  Thanks!

**Pull request merged**